src/pages/Products/Products.js

```diff
@@ -0,0 +1,91 @@
+import "./Products.css";
+import { useEffect, useState } from "react";
+import axios from "axios";
+import ProductCard from "../../components/Card/ProductCard";
+import SideBar from "../../components/SideBar/SideBar";
+import { useFilter } from "../../context/filter-context";
+
+const Products = () => {
+  const [productsData, setProductsData] = useState([]);
+  const [filteredData, setFilteredData] = useState([]);
+  const { filterState, filterDispatch } = useFilter();
+
+  useEffect(() => {
+    (async () => {
+      const data = await axios
+        .get("/api/products")
+        .then((res) => res.data.products);
+      setProductsData(data);
+      setFilteredData(data);
+    })();
+  }, []);
+
+  useEffect(() => {
+    //filterByCategory
+    let categoryData = productsData;
+    function filterCategory(category) {
+      if (filterState.filterByCategory.length === 0) return true;
+      let res = false;
+      for (let i = 0; i < filterState.filterByCategory.length; i++) {
+        res |= category === filterState.filterByCategory[i];
+      }
+      return res;
+    }
+    categoryData = categoryData.filter((item) => filterCategory(item.category));
+
+    //filterByBrand
+    let xdata = categoryData;
```

Comment:
Try to use meaningful names for the variables.

Reply:
ok sure